File: slp/backend-dotnet/Features/Dashboard/StaticWordOfTheDayProvider.cs
using backend_dotnet.Features.Dashboard;

public class StaticWordOfTheDayProvider : IWordOfTheDayProvider
{
    private static readonly List<WordOfTheDayDto> _words = new()
    {
        new() {
            Word = "perspicacious",
            PartOfSpeech = "adjective",
            VietnameseTranslation = "sắc sảo",
            Example = "She is a perspicacious student.",
            Origin = "From Latin perspicax, from perspicere 'look closely'.",
            FunFact = "First used in English in the 1630s."
        },
        new() {
            Word = "ephemeral",
            PartOfSpeech = "adjective",
            VietnameseTranslation = "phù du, ngắn ngủi",
            Example = "Social media fame is often ephemeral.",
            Origin = "From Greek ephēmeros 'lasting only a day'.",
            FunFact = "Mayflies are called ephemeroptera because of their short lifespan."
        },
        // Add more words as needed...
    };

    public Task<WordOfTheDayDto> GetWordOfTheDayAsync()
    {
        var dayOfYear = DateTime.UtcNow.DayOfYear;
        var index = (dayOfYear - 1) % _words.Count;
        return Task.FromResult(_words[index]);
    }
}

File: slp/backend-nestjs/src/modules/dashboard/providers/static-word-of-the-day.provider.ts
import { Injectable } from '@nestjs/common';
import { IWordOfTheDayProvider } from '../interfaces/word-of-the-day-provider.interface';
import { WordOfTheDayDto } from '../dto/word-of-the-day.dto';

const words: WordOfTheDayDto[] = [
  {
    word: 'perspicacious',
    partOfSpeech: 'adjective',
    vietnameseTranslation: 'sắc sảo',
    example: 'She is a perspicacious student.',
    origin: 'From Latin perspicax, from perspicere "look closely".',
    funFact: 'First used in English in the 1630s.',
  },
  {
    word: 'ephemeral',
    partOfSpeech: 'adjective',
    vietnameseTranslation: 'phù du, ngắn ngủi',
    example: 'Social media fame is often ephemeral.',
    origin: 'From Greek ephēmeros "lasting only a day".',
    funFact: 'Mayflies are called ephemeroptera because of their short lifespan.',
  },
  // Add more as needed
];

@Injectable()
export class StaticWordOfTheDayProvider implements IWordOfTheDayProvider {
  async getWordOfTheDayAsync(): Promise<WordOfTheDayDto> {
    const dayOfYear = this.getDayOfYear(new Date());
    const index = (dayOfYear - 1) % words.length;
    return words[index];
  }

  private getDayOfYear(date: Date): number {
    const start = new Date(date.getFullYear(), 0, 0);
    const diff = date.getTime() - start.getTime();
    const oneDay = 1000 * 60 * 60 * 24;
    return Math.floor(diff / oneDay);
  }
}